src/node/biquad_filter.rs

```diff
@@ -0,0 +1,1180 @@
+                .create_audio_param(g_param_opts, registration.id());
+
+            g_param.set_value(g_value);
+
+            let type_ = Arc::new(AtomicU32::new(t_value as u32));
+
+            let inits = Params {
+                q: q_value,
+                detune: d_value,
+                frequency: f_value,
+                gain: g_value,
+                type_: t_value,
+            };
+
+            let (sender, receiver) = crossbeam_channel::bounded(0);
+
+            let config = RendererConfig {
+                sample_rate,
+                gain: g_proc,
+                detune: d_proc,
+                frequency: f_proc,
+                q: q_proc,
+                type_: type_.clone(),
+                params: inits,
+                receiver,
+            };
+
+            let renderer = BiquadFilterRenderer::new(config);
+            let node = Self {
+                sample_rate,
+                registration,
+                channel_config: options.channel_config.into(),
+                type_,
+                q: q_param,
+                detune: d_param,
+                frequency: f_param,
+                gain: g_param,
+                sender,
+            };
+
+            (node, Box::new(renderer))
+        })
+    }
+
+    /// Returns the gain audio paramter
+    #[must_use]
+    pub const fn gain(&self) -> &AudioParam {
+        &self.gain
+    }
+
+    /// Returns the frequency audio paramter
+    #[must_use]
+    pub const fn frequency(&self) -> &AudioParam {
+        &self.frequency
+    }
+
+    /// Returns the detune audio paramter
+    #[must_use]
+    pub const fn detune(&self) -> &AudioParam {
+        &self.detune
+    }
+
+    /// Returns the Q audio paramter
+    #[must_use]
+    pub const fn q(&self) -> &AudioParam {
+        &self.q
+    }
+
+    /// Returns the biquad filter type
+    #[must_use]
+    pub fn type_(&self) -> BiquadFilterType {
+        self.type_.load(Ordering::SeqCst).into()
+    }
+
+    /// biquad filter type setter
+    ///
+    /// # Arguments
+    ///
+    /// * `type_` - the biquad filter type (lowpass, highpass,...)
+    pub fn set_type(&mut self, type_: BiquadFilterType) {
+        self.type_.store(type_ as u32, Ordering::SeqCst);
+    }
+
+    /// Returns the frequency response for the specified frequencies
+    ///
+    /// # Arguments
+    ///
+    /// * `frequency_hz` - frequencies for which frequency response of the filter should be calculated
+    /// * `mag_response` - magnitude of the frequency response of the filter
+    /// * `phase_response` - phase of the frequency response of the filter
+    #[allow(clippy::cast_possible_truncation)]
+    pub fn get_frequency_response(
+        &self,
+        frequency_hz: &mut [f32],
+        mag_response: &mut [f32],
+        phase_response: &mut [f32],
+    ) {
+        self.validate_inputs(frequency_hz, mag_response, phase_response);
+        let (sender, receiver) = crossbeam_channel::bounded(0);
+        self.sender
```

Comment:
I am not able to send message with OfflineAudioContext. It is the correct behavior or it is a bug ?

Reply:
It's tricky. The OfflineAudioContext is inactive until you call `start_rendering`. So putting any messages in the rendezvouz channel will hang the main thread since there is no rendering thread picking up the input.

An option would be to let `start_rendering` start a separate thread, and return a `JoinHandle<AudioBuffer>`. This is also more spec compliant. Please do not do this in this PR though, we can make a separate issue